ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/PlanModifierForASTConv.java

```diff
@@ -71,7 +73,10 @@ public class PlanModifierForASTConv {
   public static RelNode convertOpTree(RelNode rel, List<FieldSchema> resultSchema, boolean alignColumns)
       throws CalciteSemanticException {
     if (rel instanceof HiveValues) {
-      return rel;
+      RelDataTypeFactory typeFactory = rel.getCluster().getTypeFactory();
+      List<String> fieldNames = resultSchema.stream().map(FieldSchema::getName).collect(Collectors.toList());
+      fieldNames = SqlValidatorUtil.uniquify(fieldNames, typeFactory.getTypeSystem().isSchemaCaseSensitive());
```

Comment:
Is it necessary to create uniq field names? I am asking cause it seems that due to this the CLI header in some existing tests change.

Reply:
The new column names are the same as a non empty query's result which top operator is not `HiveValues`. Example:
```
select * from (select a, b from t1) s
left join t2 on s.a = t2.c;
```
```
select * from (select a, b from t1 where 0 = 1) s
left join t2 on s.a = t2.c;
```
both produces column names:
```
s.a	s.b	t2.c	t2.d
```

Reply:
OK, so the fact that we had `a b c d` was a bug right?

Reply:
Yes.